panel/widgets/slider.py

```diff
@@ -191,14 +264,38 @@ def _process_property_change(self, msg):
 
 
 class DiscreteSlider(CompositeWidget, _SliderBase):
+    """The DiscreteSlider widget allows selecting a value from a discrete list or dictionary of
+    values using a slider.
+
+    Reference: https://panel.holoviz.org/reference/widgets/DiscreteSlider.html
 
-    options = param.ClassSelector(default=[], class_=(dict, list))
+    ## Example
 
-    value = param.Parameter()
+    ```python
+    import panel as pn
+
+    pn.extension()
+
+    slider = pn.widgets.DiscreteSlider(
+        value="love",
+        options=list(["We", "love", "data", "apps"]),
+        name="A discrete value",
+    )
+    ```
+    """
+    value = param.Parameter(doc="""
+        The selected value of the slider. Updated when the handle is dragged. Must be one of
+        the options.""")
 
-    value_throttled = param.Parameter(constant=True)
+    options = param.ClassSelector(default=[], class_=(dict, list), doc="""
+        A list or dictionary of valid options.""")
 
-    formatter = param.String(default='%.3g')
+    value_throttled = param.Parameter(constant=True, doc="""The value of the slider. Updated when
+        the handle is released""")
+
+    # Why do we have both a format and formatter parameter?
```

Comment:
Just adding a comment here for now so this is not going to be lost.

Reply:
Thanks